Data/Repositories/CandidateRepository.cs

```diff
@@ -1,14 +1,56 @@
-﻿
+﻿using Data.Entities;
+
 namespace Data.Repositories
 {
-    public class CandidateRepository 
+    public class CandidateRepository : BaseRepository<Candidate>
     {
-        public void GetCandidatetById(int id) { }
+        public CandidateRepository(DataContext context) : base(context)
+        {
+
+        }
+
+        public Candidate GetCandidateById(int id) {
+            Candidate result = GetById(id);
+
+            if (result == default) return null;
```

Comment:
default for Candidate is null. If result is null you return null. Just return result, it the same logic